src/Parse/Contracts/DataFileInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php namespace Winter\Storm\Parse\Contracts;
+
+interface DataFileInterface
+{
+    /**
+     * Return a new instance of `DataFileInterface` ready for modification of the provided filepath.
+     */
+    public static function open(string $filePath): ?DataFileInterface;
+
+    /**
+     * Set a property within the data.
+     *
+     * @param string|array $key
+     * @param mixed|null $value
```

Comment:
@jaxwilko Since we're targeting PHP 8, you can use union types for the parameters now ;)